### PR TITLE
Dockerfile: Remove and Reduce unused packages

### PR DESCRIPTION
**Pull request by yukiisbored**
espeak is removed because the associated linter is removed, see https://github.com/coala/coala/issues/1939.

Mono is removed because CSharpLintBear is broken, see https://github.com/coala/docker-coala-base/issues/65

OpenJDK development packages is replaced with a smaller JRE package because PMD and Checkstyle doesn't require JDK.

ruby-common is removed because ruby and ruby-devel is more than enough.

With all of these changes, we've reduced around ~113 MB

### CODE REVIEW

**Review by robbyoconnor — CHANGES_REQUESTED** [review]
Why not change the basd image to Alpine Linux?

**Comment by jayvdb** [comment]
iirc gfortran is needed for R lintr.
bzr and hg are included so people can use VCS to fetch code to be checked.

**Comment by yukiisbored** [comment]
@robbyoconnor requires a pretty massive change, see https://travis-ci.org/yukiisbored/docker-coala-base/builds/185973853. It generates a bigger image than openSUSE but that's because I just moved it to Alpine without tweaking. Here's another attempt of redoing the docker image into Alpine https://travis-ci.org/yukiisbored/docker-coala-base/builds/186022727. 

**EDIT**: Alpine Linux is not a good choice because some linters like dart requires glibc, **not** musl. Alpine Linux also doesn't have a good way of obtaining a working and minimal Haskell setup, `ghc-bootstrap` is 1 GB. I don't think compiling those from source can do a good job in making the docker image smaller since we're adding more development tools and libraries to compile them and not to mention having a good Haskell setup is pretty hard. The other way is to use `mitchty/alpine-ghc` as the base but that doesn't solve the problem with dart and possibly other linters as well. If we have to migrate away from openSUSE, the other good choice is Ubuntu or Arch, but that would require a really really huge change (I tried this for fun, see my fork and look at the `diet-coala-ubuntu` and `diet-coala-arch` branch)

**TL;DR** Alpine is bad for this because it uses musl, not glibc

**Comment by yukiisbored** [comment]
@jayvdb I don't think lintr requires gfortran since there's no fortran code and there's no mention of `fortran` anywhere.

**Comment by jayvdb** [comment]
If it isnt used, why is espeak installed in the CI for coala-bears?

Likewise for R, but see also https://travis-ci.org/coala/docker-coala-base/builds/186067098#L5707

**Comment by jayvdb** [comment]
Removing mono should be a separate commit, closing the open issue about that.

The other changes look interesting.
How much size does this remove?

**Comment by yukiisbored** [comment]
113 MB

**Comment by jayvdb** [comment]
> If it isnt used, why is espeak installed in the CI for coala-bears?

You were right about espeak.  See https://github.com/coala/coala-bears/pull/1188
Add another commit removing that (and again calculate how much this PR should have reduced the image size).

**Comment by jayvdb** [comment]
ack 6dcf360
ack a533f9a

**Comment by jayvdb** [comment]
unack 6dcf360

**Comment by jayvdb** [comment]
Replacing `R-base` with `R-core` + `R-core-packages` is not having the effect described in the commit message.  The R packages installed are identical, probably due to cross-package "recommends" or something similar.
If you cant figure this R deps stuff, leave the R dependencies alone (and remove from the commit message).
The other removals that you have found are significant enough to be helpful for our end-users.  Thanks.

**Comment by yukiisbored** [comment]
@jayvdb I've removed espeak and added how much this PR will reduce the image size.

**Comment by yukiisbored** [comment]
@jayvdb I've made espeak into it's own commit

**Comment by jayvdb** [comment]
ack 0253896
ack 4531b7a
ack 6c7c6c1

**Comment by jayvdb** [comment]
@rultor merge
